fastcrypto-vdf/src/vdf/pietrzak/mod.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::math::parameterized_group::{multiply, ParameterizedGroupElement};
+use crate::vdf::VDF;
+use fastcrypto::error::FastCryptoError::{InvalidInput, InvalidProof};
+use fastcrypto::error::FastCryptoResult;
+use fastcrypto::hash::{HashFunction, Keccak256};
+use num_bigint::BigUint;
+use num_integer::Integer;
+use serde::Serialize;
+use std::mem;
+
+/// Default size in bytes of the Fiat-Shamir challenge used in proving and verification.
+pub const DEFAULT_CHALLENGE_SIZE_IN_BYTES: usize = 32;
+
+/// This implements Pietrzak's VDF construction from https://eprint.iacr.org/2018/627.pdf.
+/// Proofs are larger and verification is slower than in Wesolowski's construction, but the
+/// output of a VDF is unique, assuming that the used group have no small subgroups, and proving
```

Comment:
what does it mean that proving is faster - for the same security? or, for the same number of iterations (which implies weaker security)?

Reply:
Yes, it's for the same number of iterations. Note that evaluation of the VDF is the part that should be linear in the number of iterations and proving should just be faster than evaluation, so it will not impact security.